packages/cli/src/services/public-api-key.service.ts

```diff
@@ -0,0 +1,84 @@
+import { randomBytes } from 'crypto';
+import Container, { Service } from 'typedi';
+
+import { ApiKey } from '@/databases/entities/api-key';
+import type { User } from '@/databases/entities/user';
+import { ApiKeyRepository } from '@/databases/repositories/api-key.repository';
+import { UserRepository } from '@/databases/repositories/user.repository';
+
+export const API_KEY_PREFIX = 'n8n_api_';
+
+@Service()
+export class PublicApiKeyService {
+	constructor(private readonly apiKeysRepository: ApiKeyRepository) {}
+
+	/**
+	 * Creates a new public API key for the specified user.
+	 *
+	 * @param {User} user - The user for whom the API key is being created.
+	 * @returns {Promise<ApiKey>} A promise that resolves to the newly created API key.
```

Comment:
We don't necessarily need to annotate the types, as we use typescript. It's duplication basically. Rather specify the return type in TS